tensorflow/python/keras/layers/dense_attention.py

```diff
@@ -192,10 +192,10 @@ def _validate_call_args(self, inputs, mask):
         raise ValueError(
             '{} layer mask must be a list, '
             'namely [query_mask, value_mask].'.format(class_name))
-      if len(mask) != 2:
+      if len(mask) < 2 or len(mask) > 3:
         raise ValueError(
-            '{} layer mask must be a list of length 2, namely [query_mask, '
-            'value_mask]. Given length: {}'.format(class_name, len(mask)))
+            '{} layer mask must be a list of length 2 or 3, namely [query_mask, '
+            'value_mask] or [query_mask, value_mask, key_mask]. Given length: {}'.format(class_name, len(mask)))
```

Comment:
Please don't change it to include key_mask, since we don't expect user to provide it, and will not be used even provided.

Reply:
My intent was for it not to error when 3 masks are passed via implicit propagation as in the example below. Should I remove the changes to the ValueError() while still letting it accept 3 arguments or just close this pull request?

```
input = tf.keras.layers.Input(shape=(None,))
x = tf.keras.layers.Embedding(input_vocab_size, d_model, mask_zero=True)(input)
query = tf.keras.layers.Dense(d_model)(x)
value =  tf.keras.layers.Dense(d_model)(x)
key    =  tf.keras.layers.Dense(d_model)(x)
attention = tf.keras.layers.Attention()([query, value, key])
...
```

Reply:
I think you can keep the check for len(mask) in (2, 3), but change the error message back to only list the query_mask and value_mask. We don't want to encourage user to explicitly pass in the key_mask, which they can do like:

```
attention = tf.keras.layers.Attention()([query, value, key], 
    masks=[query._keras_mask, value._keras_mask, key._keras_mask])
```